### PR TITLE
storage: etcd: use sub-tests

### PR DESCRIPTION
**Pull request by stevekuznetsov**
Sub-tests are much friendlier to anyone breaking these tests than the old-school methods.

Signed-off-by: Steve Kuznetsov <skuznets@redhat.com>

/kind cleanup
/sig api-machinery
/cc @stts @liggitt 

```release-note
NONE
```

```docs

```

### CODE REVIEW

File: staging/src/k8s.io/apiserver/pkg/storage/etcd3/store_test.go [hunk 1]
```diff
@@ -300,34 +300,39 @@ func TestUnconditionalDelete(t *testing.T) {
...
-			continue
-		}
-		if err != nil {
-			t.Fatalf("Delete failed: %v", err)
-		}
-		if !reflect.DeepEqual(tt.expectedObj, out) {
-			t.Errorf("#%d: pod want=%#v, get=%#v", i, tt.expectedObj, out)
-		}
+			if err != nil {
+				t.Fatalf("#%d: %s: Delete failed: %v", i, tt.name, err)
```
Comment (liggitt) [inline staging/src/k8s.io/apiserver/pkg/storage/etcd3/store_test.go:330]: no need for `i` inside a t.Run

File: staging/src/k8s.io/apiserver/pkg/storage/etcd3/store_test.go [hunk 2]
```diff
@@ -300,34 +300,39 @@ func TestUnconditionalDelete(t *testing.T) {
...
 		expectNotFoundErr bool
-	}{{ // test unconditional delete on existing key
+	}{{
+		name: "test unconditional delete on existing key",
 		key:               key,
 		expectedObj:       storedObj,
 		expectNotFoundErr: false,
-	}, { // test unconditional delete on non-existing key
+	}, {
+		name: "test unconditional delete on non-existing key",
```
Comment (liggitt) [inline staging/src/k8s.io/apiserver/pkg/storage/etcd3/store_test.go:313]: names passed to `t.Run` show up in test grid and explode the left-hand side width, right? might need to trim verbosity a little/a lot

Comment (stevekuznetsov) [reply staging/src/k8s.io/apiserver/pkg/storage/etcd3/store_test.go:313]: @liggitt I'm not extremely familiar with TestGrid rendering, sorry - is there a guideline or length limit I should target? I can take a stab at shortening the names.

File: staging/src/k8s.io/apiserver/pkg/storage/etcd3/store_test.go (hunk 1 above)
Comment (stevekuznetsov) [reply staging/src/k8s.io/apiserver/pkg/storage/etcd3/store_test.go:330]: ACK - I was looking to make the smallest change, but I can certainly remove the index prefix from these log lines.

File: staging/src/k8s.io/apiserver/pkg/storage/etcd3/store_test.go (hunk 2 above)
Comment (liggitt) [reply staging/src/k8s.io/apiserver/pkg/storage/etcd3/store_test.go:313]: https://testgrid.k8s.io/presubmits-kubernetes-blocking#pull-kubernetes-unit&include-filter-by-regex=k8s.io%2Fapiserver%2Fpkg%2Fstorage%2Fetcd3 is the ones we currently have

looks like we already have some ~verbose ones, so maybe this one is fine, but some of the ones below that tell little stories should be trimmed ("test GetToList on existing key with minimum resource version set to current resource version, match=minimum")

Comment (stevekuznetsov) [reply staging/src/k8s.io/apiserver/pkg/storage/etcd3/store_test.go:313]: Yeah, looks like the same comment -> name approach was taken in e.g. `TestList` conversion. I'll add a commit to shorten some of these where I can.

**Comment by stevekuznetsov** [comment]
@liggitt PTAL - was able to shorten them all, removed redundant prefixes and changed stories to explicit things (e.g. "minimum resource version set to current resource version" became "resourceVersion=current")

**Comment by stevekuznetsov** [comment]
@liggitt sorry, looks like my editor was being goofy with `gofmt`-on-save, needs another LGTM please

**Pull request merged**